### ISSUE TITLE
Profile input pipeline, or write infrastructure to pre-generate training

### ISSUE THREAD

**Issue by gatoatigrado** [aimalz/justice#102]
Pull request #101 starts building synthetic positives and negatives input pipelines, supporting train/tune splits, and for the positives mixing together test_set and training_set.

We can either to down the route of,

 * Making the input pipeline fast enough to stream examples during training.
 * Pre-generate input examples into a file with `tf.train.Example`s.

For the first option, probably the best option is to `pip install snakeviz` and `%%snakeviz` in an iPython cell, that runs example generation up until the point where we stuff things in a `tf.data.Dataset`. (there's probably a way to get the python profilers to follow threads and profile within TF calls, but it might not be worth the effort.)

For the second option, #101 has a bit of `start_seed` logic at least for positives generation, so we use `sbatch` (distributed computing) to generate examples, passing some notion of worker/job ID as `start_seed`. We'll need to extend the seed logic to negatives generation (I'll open a ticket).
